tests/test_nginx.py

```diff
@@ -10,6 +14,15 @@ def test_nginx_welcome_page(auto_container, host):
     """test that the default welcome page is served by the container."""
     host_port = auto_container.forwarded_ports[0].host_port
 
-    assert "Welcome to nginx" in host.check_output(
-        f"curl -sf --retry 5 --retry-connrefused http://localhost:{host_port}/"
+    # Retry 5 times with quadratic backoff delay
+    @retry(
+        wait=wait_exponential(multiplier=1, min=4, max=10),
+        stop=stop_after_attempt(5),
     )
+    def check_nginx_response():
+        resp = requests.get(f"http://localhost:{host_port}/", timeout=30)
+        resp.raise_for_status()
+        if "Welcome to nginx" not in resp.text:
+            raise ValueError("Invalid response")
```

Comment:
This still needs to be an assert I think?

Reply:
Don't think so - the exception should trigger a retry with tenacity, if we `assert` here then the test run would immediately fail instead of retry to fetch the contents.

I tried yesterday to get `http://localhost:12` in line 23 and the test run failed as expected.

Reply:
which is what it is supposed to do. it should only retry on connection refused, not on "successfully serving unexpected content"

Reply:
Now I understand you. Yes that makes more sense, PR amended.